src/components/@shared/button/CustomButton.tsx

```diff
@@ -20,58 +21,44 @@ export default function CustomButton({
   onClick,
   type = 'button',
   style = 'primary',
+  secondaryBg = 'bg-body-bg',
 }: PropsWithChildren<CustomButtonProps>) {
   /**active에 boolean값을 넣어주세요
-   *기본 style은 "primary"
+기본 style은 "primary"
+secondary 스타일을 사용할 때 버튼 색상이 body의 색상과 다르면 tailwind속성을 넣어주세요
    */
   return (
-    <>
-      <Button
-        type={type}
-        onClick={active ? onClick : undefined} // active가 true일 때만 실행
-        disabled={!active}
-        className={classNames(
-          'h-[50px] w-full focus:ring-0 md:h-[55px] xl:h-[65px]',
-          active ? 'bg-gradation' : 'cursor-not-allowed bg-unactive enabled:hover:bg-unactive',
-        )}
-      >
-        {style === 'primary' && <span className="flex h-full items-center text-lg font-semibold">{children}</span>}
-        {style === 'secondary' && (
-          <span
-            className={classNames(
-              'absolute inset-0.5 flex items-center justify-center rounded-lg bg-body-bg text-transparent',
-              active ? 'hover:bg-transparent hover:text-white' : '',
-            )}
-          >
-            <span
-              className={classNames(
-                'text-lg font-semibold',
-                active ? 'bg-gradation bg-clip-text' : 'text-brand-gray-dark',
-              )}
-            >
-              {children}
-            </span>
-          </span>
-        )}
-      </Button>
-      {style === 'tertiary' && (
-        <Button
-          onClick={active ? onClick : undefined}
+    <button
+      type={type}
+      onClick={active ? onClick : undefined}
+      className={classNames(
+        'relative flex h-[50px] w-full items-center justify-center rounded-lg border text-lg font-semibold md:h-[55px] xl:h-[65px]',
+        style === 'primary'
+          ? active
+            ? 'bg-gradation text-white'
+            : 'bg-unactive text-brand-gray-dark'
+          : style === 'secondary'
+            ? active
+              ? 'bg-gradation'
+              : 'border-solid border-unactive text-brand-gray-dark'
+            : active // style === "tertiary"일 때
+              ? 'border-solid border-brand-gray-light'
+              : 'border-solid border-unactive text-brand-gray-dark',
+        !active && 'cursor-not-allowed',
+      )}
+    >
```

Comment:
삼항연산자가 많이 쓰여서 가독성이 떨어질 수 있는데 주강사님이 항상 추천하시는 `ts-pattern` 라이브러리를 적용시켜 볼 수도 있을 거 같아용!
아래는 지피티에게 간단히 적용시켜 달라고 물어봐서 얻은 답변 코드입니다
```typescript
import { match } from 'ts-pattern';

export default function CustomButton({
  type,
  onClick,
  active,
  style,
}: {
  type: string;
  onClick: () => void;
  active: boolean;
  style: 'primary' | 'secondary' | 'tertiary';
}) {
  const classNames = match({ style, active })
    .with({ style: 'primary', active: true }, () => 'bg-gradation text-white')
    .with({ style: 'primary', active: false }, () => 'bg-unactive text-brand-gray-dark')
    .with({ style: 'secondary', active: true }, () => 'bg-gradation')
    .with({ style: 'secondary', active: false }, () => 'border-solid border-unactive text-brand-gray-dark')
    .with({ style: 'tertiary', active: true }, () => 'border-solid border-brand-gray-light')
    .with({ style: 'tertiary', active: false }, () => 'border-solid border-unactive text-brand-gray-dark')
    .otherwise(() => '');

  return (
    <button
      type={type}
      onClick={active ? onClick : undefined}
      className={classNames.concat(
        'relative flex h-[50px] w-full items-center justify-center rounded-lg border text-lg font-semibold md:h-[55px] xl:h-[65px]',
        !active ? ' cursor-not-allowed' : ''
      )}
    >
      {/* 버튼 내용 */}
    </button>
  );
}
```

Reply:
사용법이 약간 달라서 찾아보고 수정했습니다!